components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2254,10 +2254,28 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
             const billingMode = await this.billingModes.getBillingModeForUser(user, new Date());
             if (billingMode.mode === "usage-based") {
                 const limit = await this.billingService.checkUsageLimitReached(user);
-                if (limit.reached) {
-                    result.unshift("The usage limit is reached.");
-                } else if (limit.almostReached) {
-                    result.unshift("The usage limit is almost reached.");
+                let teamOrUser;
+                switch (limit.attributionId.kind) {
+                    case "user": {
+                        if (limit.reached) {
+                            result.unshift(`You have reached your usage limit.`);
+                        } else if (limit.almostReached) {
+                            result.unshift(`You have almost reached your usage limit.`);
+                        }
+                        break;
+                    }
+                    case "team": {
+                        teamOrUser = await this.teamDB.findTeamById(limit.attributionId.teamId);
+                        if (teamOrUser) {
+                            result.push(teamOrUser?.slug);
+                            if (limit.reached) {
+                                result.unshift(`Your team '${teamOrUser?.name}' has reached its usage limit.`);
+                            } else if (limit.almostReached) {
+                                result.unshift(`Your team '${teamOrUser?.name}' has almost reached its usage limit.`);
+                            }
+                        }
+                        break;
+                    }
```

Comment:
The nesting here makes it quite hard to follow the main flow of the method. What if we refactored this block into a separate method and do something like 
```
const usageNotification = getUsageNotificationForAttributionID(...)
if usageNotification.length > 0 { ... }
```

Reply:
Sorry @easyCZ I was refactoring this but forgot to hold and now it's merged!